app/uk/gov/hmrc/fileupload/write/envelope/Envelope.scala

```diff
@@ -213,7 +234,36 @@ sealed trait State {
         Xor.left(FileAlreadyProcessed)
       }).getOrElse(fileNotFoundError)
 
-  def checkCanStoreFile(fileId: FileId, fileRefId: FileRefId, files: Map[FileId, File]): CanResult =
+  def checkCanFileQuarantined(fileId: FileId, fileRefId: FileRefId, fileLength: Long, files: Map[FileId, File], envelope: Envelope): CanResult = {
+
+    def sizeToByte(size: String): Long = {
+      val sizeRegex = "([1-9][0-9]{0,3})([KB,MB]{2})".r
```

Comment:
Actually after some consideration you actually do reject as part of creating the envelope if values are incorrect so I was probably to eager to criticise here.